flow/activities/flowable.go

```diff
@@ -834,7 +834,21 @@ func (a *FlowableActivity) QRepWaitUntilNewRows(ctx context.Context,
 	attemptCount := 1
 	for {
 		activity.RecordHeartbeat(ctx, fmt.Sprintf("no new rows yet, attempt #%d", attemptCount))
-		time.Sleep(waitBetweenBatches)
+		waitUntil := time.Now().Add(waitBetweenBatches)
```

Comment:
Maybe this can be made into a utility function in `connectors/utils/heartbeat.go`